rfc/005-soft-chain-upgrades.md

```diff
@@ -0,0 +1,473 @@
+
+# RFC 005: Soft Chain Upgrades
+
+## Changelog
+
+- 2020-11-27: Add appendix to distinguish between soft and hard upgrades
+- 2020-11-23: Added chain migration proposal for comparison
+- 2020-11-16: Initial draft.
+
+## Author(s)
+
+- Callum Waters (@cmwaters)
+
+## Context
+
+The current method for upgrading a network across block protocol breaking
+changes is to hard fork the chain. All nodes must coordinate a halt height/
+time, where application state can be exported across and an entirely new chain
+(as far as Tendermint is concerned) is created (see [cosmoshub-3
+upgrade](https://blog.cosmos.network/cosmos-hub-3-upgrade-announcement-39c9da941aee)).
+This means that each block protocol version maps to the entire life of a single
+chain. The result is that networks either forego new features for long periods
+of time or have to deal with the pains of upgrading such as the coordination
+effort, state migration, intra-chain security challenges and most importantly,
+the loss of transaction history.
+
+The scope of this RFC is to address the need for Tendermint to have better
+flexibility in improving itself whilst offering greater ease of use to the
+networks running on Tendermint. This is done by introducing a division in the
```

Comment:
Was there a concept of soft upgrades before in tendermint? 

I'm guessing before this, operators would perform minor upgrades to their nodes within a live network (especially if it was a security fix) in their own time - so this would constitute a soft upgrade.

In this case you're right. In the pie of possible upgrades we are trying to get a bigger slice of them to not require a hard reset.